app/controllers/invoices_controller.rb

```diff
@@ -85,6 +85,21 @@ def update
       render :edit
     end
   end
+
+  def duplicate
+    @old_invoice = Invoice.includes(:event, :invoice_lines).find(params[:id])
+    @invoice = @old_invoice.amoeba_dup
+    
+    authorize! :create, @invoice
+
+    if @invoice.save
+      flash[:notice] = "Invoice duplicated successfully!"
+    else
+      flash[:alert] = "Invoice duplicated unsuccessfully!"
```

Comment:
As a phrase, "invoice duplicated" kind of indicates success, so using it in the failure message is a bit confusing.  Consider instead a matter-of-fact "Error duplicating invoice."  Also should be a `flash[:error]` instead of a `flash[:alert]` for consistency with [similar errors](https://github.com/ABTech/abtt/search?utf8=%E2%9C%93&q=flash+%28alert+OR+error%29&type=).